### ISSUE TITLE
Use new syntax for creating multiple references in QuickStatements export

### ISSUE THREAD

**Issue by wetneb** [OpenRefine/OpenRefine#6149]
[QuickStatements](https://www.wikidata.org/wiki/Help:QuickStatements) has [introduced](https://www.wikidata.org/w/index.php?title=Help%3AQuickStatements&diff=prev&oldid=1541064004) a new syntax to let people add multiple references to a statement, using exclamation marks to separate distinct references.

OpenRefine's Wikibase extension should be updated to use this syntax.

### Proposed solution

When creating a statement with multiple references, instead of creating a QuickStatements script of the form
```
Qaaa Pbbb Qccc Sddd Qeee
Qaaa Pbbb Qccc Sfff Qggg
```
we should rather output a script of the form
```
Qaaa Pbbb Qccc Sddd Qeee !Sfff Qggg
```

### Alternatives considered
None

### Additional context

Reported on the OpenRefine-Wikimedia telegram Group by @SannitaSSJ

**Comment by payalsaraljain** [comment]
Hi can I take this issue?

**Comment by payalsaraljain** [comment]
Hi, I am interested in checking out this issue, it is assigned to someone?

**Comment by wetneb** [comment]
Amazing, it is yours @payalsaraljain :)

**Comment by payalsaraljain** [comment]
Thank you, I will start looking into this

**Comment by payalsaraljain** [comment]
Quick question there should be no tab space between the ! and reference right?

**Comment by wetneb** [comment]
Yes, that's also my understanding of [the docs](https://www.wikidata.org/wiki/Help:QuickStatements#Add_statement_with_sources). You could double-check that by making some sample edits on the [Wikidata Sandbox](https://www.wikidata.org/wiki/Q4115189) via QuickStatements, trying to add multiple references at once.

**Comment by payalsaraljain** [comment]
Hi, can you elaboratea little on how to do that I am unable to do it

**Comment by wetneb** [comment]
Sure. First, you'll need to create a Wikidata account. It's useful if you learn how to edit Wikidata manually first, for instance by following the( Wikidata tours](https://www.wikidata.org/wiki/Wikidata:Tours). That will help you get familiar with the different parts of Wikidata's data model (importantly, for this issue: items, properties, statements and references).

Then, you need to learn how to make Wikidata edits via [QuickStatements](https://quickstatements.toolforge.org/#/). To do so, you could try to follow [the docs](https://www.wikidata.org/wiki/Help:QuickStatements) or watch [some video tutorial](https://www.youtube.com/watch?v=L0TYQ9LRRTQ).

Once you have made a few edits with QuickStatements, you could experiment with the specific syntax that is available to create statements with multiple references. You should try and craft a small QuickStatements script that uses this syntax (using the `!` character). Because you will be making a test edit, I recommend that the subject Wikidata item you use is a so-called "sandbox item", which is a particular Wikidata item where it's okay to make gibberish edits. One such item is https://www.wikidata.org/wiki/Q4115189

Check that your script works by running it in QuickStatements. If it does not create two references as expected, you could try getting some help, for instance by starting a new discussion over at https://www.wikidata.org/wiki/Help_talk:QuickStatements.

If your script does work, then you should be ready to implement support for this syntax in OpenRefine. For that part, you should be able to follow this guide: https://openrefine.org/docs/technical-reference/code-contributions

**Comment by payalsaraljain** [comment]
Hi @wetneb, for running edits via Quick statements, it gives me "You can't create a new batch, because you are not autoconfirmed", on probing on this I found that this does not work for newly created Wikidata account, is there any other way for validating?

**Comment by payalsaraljain** [comment]
Hi @wetneb, I followed the steps, up until making few edits via QuickStatements,  it is saying You can't create a new batch, because you are not autoconfirmed, on probing on this I found that it has some restriction on running certain commads for newly created wikidata account, if that is the case is there another way of validating?

**Comment by wetneb** [comment]
That's a good point! You could either do a few more edits to reach the autoconfirmed status (https://www.wikidata.org/wiki/Wikidata:Autoconfirmed_users) or run your test statements on another wiki. For instance, there are a number of test Wikibase instances listed at https://www.wikibase.cloud/discovery, such as https://naf-test.wikibase.cloud/wiki/NAF_Test or https://metadata-test2023-rbx.wikibase.cloud/wiki/Main_Page. You could try using one of them (and reverting your edits afterwards)

**Comment by wetneb** [comment]
@payalsaraljain I am assuming you stopped working on this by now, so I am de-assigning you from this for now. If you are still working on this, let me know what is holding you back and I'll happily re-assign you.

**Comment by Ahmed-Elgamel** [comment]
Hi @wetneb can you assign me this issue?

**Comment by Ahmed-Elgamel** [comment]
Hi @wetneb , I have gone through the [Wikidata tours](https://www.wikidata.org/wiki/Wikidata:Tours) and made a Wikidata account.I am doing edits on a test Wikibase instance as you mentioned earlier(I am not auto confirmed on WikiData yet).I am trying to enter the command that creates a statement with multiple references but it does not seem to understand this syntax and behaves unexpectedly.

The command was worked correctly when we added multiple references in the same group
<img width="1058" alt="Screenshot 2024-03-11 at 1 25 59 AM" src="https://github.com/OpenRefine/OpenRefine/assets/116679740/09a777f2-e22c-4779-8d49-8b86e5e0a2f8">
<img width="1488" alt="Screenshot 2024-03-11 at 1 26 12 AM" src="https://github.com/OpenRefine/OpenRefine/assets/116679740/098233ab-e551-450d-9279-aa7e4300d56d">


But when I try to put each reference in a separate group it does not recognize the exclamation mark.
<img width="1473" alt="Screenshot 2024-03-11 at 1 27 41 AM" src="https://github.com/OpenRefine/OpenRefine/assets/116679740/3c6ae73a-6b9a-40fe-b5a0-fc76fb5c18fa">
<img width="1493" alt="Screenshot 2024-03-11 at 1 27 46 AM" src="https://github.com/OpenRefine/OpenRefine/assets/116679740/ab20e7ea-c2b1-462a-ac7e-3c6fd06c0471">
In the last photo it just recognizes the first source/reference while the second one is not recognized.

Is there something that I am misunderstanding or doing wrong?

**Comment by wetneb** [comment]
Hi @Ahmed-Elgamel,

You're totally right, I also cannot get it to work on my side either. Weird! Thanks for checking.
@SannitaSSJ can you confirm how to use this syntax?

Removing the "good first issue" label in the meantime.

**Comment by wetneb** [comment]
Closing as the syntax appears not to be supported.
